lib/state/revision/reducer.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	REVISION_IS_VIEWING,
+	REVISION_SELECT,
+} from '../action-types';
+
+export const isViewingRevisions = ( state = false, { type, isViewing } ) =>
+	REVISION_IS_VIEWING === type ? isViewing : state;
+
+export const selectedRevision = ( state = null, { type, revision } ) =>
+	REVISION_SELECT === type ? revision : state;
+
+export default combineReducers( {
+	isViewingRevisions,
```

Comment:
I guess we should start with the assumption that the newest revision is... the note itself!
In the current code the object that goes into `<NoteDetail>` is even called [revision](https://github.com/Automattic/simplenote-electron/blob/master/lib/note-editor.jsx#L90), and it's the selected revision (if it exists) or the note.
While I'm not sold on the naming confusion, I dig the simplicity of the logic behind this.

So, if we want to have only one `state.revision` reducer, we could simplify it like this (and notice that I'd keep the action names starting with the feature, to improve their searchability):

**default state**
- `selectedRevision = null`
- `<RevisionSelector>` hidden

**open the revision selector**
- dispatch `REVISION_SELECT` -> `selectedRevision = note`
- `<RevisionSelector>` visible

**close the revision selector** (cancel or click outside)
- dispatch `REVISION_SELECT` -> `selectedRevision = null`
- `<RevisionSelector>` hidden

**select a revision**
- dispatch `REVISION_SELECT` -> `selectedRevision = revision`
- `<RevisionSelector>` visible

**restore a revision** (and close the revision selector)
- `note = selectedRevision`
- dispatch `REVISION_SELECT` -> `selectedRevision = null`
- `<RevisionSelector>` hidden

**change note**
- dispatch `REVISION_SELECT` -> `selectedRevision = null`
- `<RevisionSelector>` hidden

---

This way, the pattern to toggle the `<RevisionSelector>` visibility is clear:
```jsx
{ selectedRevision && <RevisionSelector> }
```
and we'll spare the toggle visibility reducer (`isViewingRevisions`), that I feel might be a little out of place in `state.ui`, since it's so strictly tied to the note editor itself.